### PR TITLE
Add SyncCacheApi implementation over PlayRedis class.

### LINKED ISSUE #6: Use Async (Sync) Cache Api to get cache content
One of the best things we have in Play is the interface to implement our own Cache Api.
I think we can leverage this advantage to use this Redis Module over Async/Sync Cache Api.

https://www.playframework.com/documentation/2.8.x/JavaCache
https://www.playframework.com/documentation/2.8.x/ScalaCache

### PR DESCRIPTION
**Pull request by felipebonezi**
Signed-off-by: Felipe Bonezi <1410319+felipebonezi@users.noreply.github.com>

Issue Ref: #6

### CODE REVIEW

**Comment by felipebonezi** [comment]
@thibaultmeyer Can I implement the `AsyncCacheAPI` in this same Pull Request?
Or is it best to create a new one?

**Comment by thibaultmeyer** [comment]
Hello,

You can implements it in the same merge request. 

Thank you

**Comment by felipebonezi** [comment]
@thibaultmeyer I've done the implementation, but I don't know why the CodeCov isn't working!
Can you check?

**Comment by felipebonezi** [comment]
> @thibaultmeyer I've done the implementation, but I don't know why the CodeCov isn't working!
> Can you check?

All checks have passed @thibaultmeyer 🎉

File: src/main/java/com/zero_x_baadf00d/play/module/redis/cache/AsyncCacheRedisImpl.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+ */
+@Singleton
+public class AsyncCacheRedisImpl implements AsyncCacheApi {
+
+    /**
+     * Keep all keys saved into cache.
+     *
+     * @since 20.11.05
+     */
+    private final Set<String> keys = new HashSet<>(100);
```
Comment (thibaultmeyer) [inline src/main/java/com/zero_x_baadf00d/play/module/redis/cache/AsyncCacheRedisImpl.java:60]: I'm not sure. This mechanism will not work if several instances of the application are running. The simplest would be, in my opinion, to use the `getConnection` method to execute the `KEYS` method which allows to find all the keys matching needed patter.

You can use a hardcoded prefix or allow user to change it in the configuration file. By example if you use the prefix `cache.`, the method `removeAll` will use Redis connection to find all keys matching `cache.*` and delete them.

Redis doc : https://redis.io/commands/keys


**Example**
```java
@Override
public CompletionStage<Done> removeAll() {
   return supplyAsync(() -> {
        this.playRedis.getConnection(0).keys("search pattern").forEach(this::remove);
        return Done.done();
    });
}
```

Comment (thibaultmeyer) [inline src/main/java/com/zero_x_baadf00d/play/module/redis/cache/AsyncCacheRedisImpl.java]: I'm not sure. This mechanism will not work if several instances of the application are running. The simplest would be, in my opinion, to use the `getConnection` method to execute the `KEYS` method which allows to find all the keys matching needed patter.

You can use a hardcoded prefix or allow user to change it in the configuration file. By example if you use the prefix `cache.`, the method `removeAll` will use Redis connection to find all keys matching `cache.*` and delete them.

Redis doc : https://redis.io/commands/keys


**Example**
```java
@Override
public CompletionStage<Done> removeAll() {
   return supplyAsync(() -> {
        this.playRedis.getConnection(0).keys("search pattern").forEach(this::remove);
        return Done.done();
    });
}
```

**Review by thibaultmeyer — CHANGES_REQUESTED** [review]

File: src/main/java/com/zero_x_baadf00d/play/module/redis/cache/AsyncCacheRedisImpl.java (hunk 1 above)
Comment (felipebonezi) [reply src/main/java/com/zero_x_baadf00d/play/module/redis/cache/AsyncCacheRedisImpl.java]: @thibaultmeyer thanks, I think it makes sense.

File: src/main/java/com/zero_x_baadf00d/play/module/redis/cache/AsyncCacheRedisImpl.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        });
+    }
+
+    @Override
+    public CompletionStage<Done> removeAll() {
+        return supplyAsync(() -> {
+            try (final Jedis connection = this.asPlayRedis().getConnection(0)) {
+                // Search all keys in Redis.
+                // https://redis.io/commands/keys
+                connection.keys(DEFAULT_PREFIX + "??").forEach(this::remove);
```
Comment (thibaultmeyer) [inline src/main/java/com/zero_x_baadf00d/play/module/redis/cache/AsyncCacheRedisImpl.java]: I think you have to replace `??` with `*`. According to the documentation `??` will only match a two characters word.

By example, if you have the following keys :
- play.cache.p1
- play.cache.cn
- play.cache.index
- play.cache.dashboard

The pattern `DEFAULT_PREFIX + "??"` will only match "play.cache.p1" and "play.cache.cn". The pattern  `DEFAULT_PREFIX + "*"` will match all keys beginning with DEFAULT_PREFIX

**Review by thibaultmeyer — CHANGES_REQUESTED** [review]

File: src/main/java/com/zero_x_baadf00d/play/module/redis/cache/AsyncCacheRedisImpl.java (hunk 2 above)
Comment (felipebonezi) [reply src/main/java/com/zero_x_baadf00d/play/module/redis/cache/AsyncCacheRedisImpl.java]: You're right!

**Review by thibaultmeyer — APPROVED** [review]
Nice work. Thank you !

**Pull request merged**